x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/googleaistudio/embeddings/GoogleAiStudioEmbeddingsServiceSettings.java

```diff
@@ -0,0 +1,189 @@
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalPositiveInteger;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractRequiredString;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractSimilarity;
+
+public class GoogleAiStudioEmbeddingsServiceSettings extends FilteredXContentObject
+    implements
+        ServiceSettings,
+        GoogleAiStudioRateLimitServiceSettings {
+
+    public static final String NAME = "google_ai_studio_embeddings_service_settings";
+
+    /**
+     * Rate limits are defined at <a href="https://ai.google.dev/pricing">Google Gemini API Pricing</a>.
+     * For pay-as-you-go you've 360 requests per minute.
+     */
+    private static final RateLimitSettings DEFAULT_RATE_LIMIT_SETTINGS = new RateLimitSettings(360);
+
+    public static GoogleAiStudioEmbeddingsServiceSettings fromMap(Map<String, Object> map) {
+        ValidationException validationException = new ValidationException();
+
+        String model = extractRequiredString(map, MODEL_ID, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        Integer maxInputTokens = extractOptionalPositiveInteger(
+            map,
+            MAX_INPUT_TOKENS,
+            ModelConfigurations.SERVICE_SETTINGS,
+            validationException
+        );
+        SimilarityMeasure similarityMeasure = extractSimilarity(map, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        Integer dims = extractOptionalPositiveInteger(map, DIMENSIONS, ModelConfigurations.SERVICE_SETTINGS, validationException);
+        RateLimitSettings rateLimitSettings = RateLimitSettings.of(map, DEFAULT_RATE_LIMIT_SETTINGS, validationException);
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
+        }
+
+        return new GoogleAiStudioEmbeddingsServiceSettings(model, maxInputTokens, dims, similarityMeasure, rateLimitSettings);
+    }
+
+    private final String modelId;
+
+    private final RateLimitSettings rateLimitSettings;
+
+    private final Integer dims;
+
+    private final Integer maxInputTokens;
+
+    private final SimilarityMeasure similarity;
+
+    public GoogleAiStudioEmbeddingsServiceSettings(
+        String modelId,
+        @Nullable Integer maxInputTokens,
+        @Nullable Integer dims,
+        @Nullable SimilarityMeasure similarity,
+        @Nullable RateLimitSettings rateLimitSettings
+    ) {
+        this.modelId = modelId;
+        this.maxInputTokens = maxInputTokens;
+        this.dims = dims;
+        this.similarity = similarity;
+        this.rateLimitSettings = Objects.requireNonNullElse(rateLimitSettings, DEFAULT_RATE_LIMIT_SETTINGS);
+    }
+
+    public GoogleAiStudioEmbeddingsServiceSettings(StreamInput in) throws IOException {
+        this.modelId = in.readString();
+        this.maxInputTokens = in.readOptionalInt();
```

Comment:
Good point! Adjusted with [Use readOptionalVInt instead of readOptionalInt](https://github.com/elastic/elasticsearch/pull/109243/commits/21d72f0f431a76a322452afe5b18be887ec7d395)